src/manager/agents/common-module/module/utils/common_utils.cpp

```diff
@@ -121,37 +165,39 @@ bool CommonUtils::TransformToScStruct(
   return true;
 }
 
-ScAddrVector CommonUtils::GetNodesUnderParameter(
+std::set<ScAddr, ScAddrLessFunc> CommonUtils::GetComponentsToInstall(
```

Comment:
```suggestion
std::unordered_set<ScAddr, ScAddrLessFunc> CommonUtils::GetComponentsToInstall(
```
Unordered set is better because we don't need to sort elements while adding to the set. It is more efficient. Check everywhere